docs/consistency.puml

```diff
@@ -0,0 +1,21 @@
+@startuml
+
+@startuml
+
+class DualConnection
+
+interface ReplicaConsistency{
```

Comment:
I wonder if build would faild if ReplicaConsistency SPI would change, but `consistency.puml` not 
@wyrzyk WDYT?

Reply:
It would be great to have such functionality, currently `consistency.puml` is not connected with the source and we need to take care of it manually.

Reply:
True, although we shouldn't change architecture that often.

https://matklad.github.io//2021/02/06/ARCHITECTURE.md.html :
> Explicitly call-out architectural invariants. Often, important invariants are expressed as an *absence* of something, and it’s pretty hard to divine that from reading the code. Nothing in the model layer specifically doesn’t depend on the views.

It's about high-level boundaries, what should go where and which parts are connected. It's not about reflecting the codebase precisely.

Reply:
> True, although we shouldn't change architecture that often.
Agree. As long as dual-connection isn't big unit/component/lib with > 15 contributors this will work.
Later ppl just come and do their best efforts, and might do some mess by accident.
With that  [fitness functions](https://www.thoughtworks.com/radar/techniques/architectural-fitness-function)
comes very healthy!


Ok I dig a little bit;
[archunit-plantuml](https://www.archunit.org/userguide/html/000_Index.html#_plantuml_component_diagrams_as_rules)
inspired by  [this googled issue](https://github.com/TNG/ArchUnit/issues/108)

It's not for now(all goods) but I wonder whether You considered such an approach for dual-connection lib? 
Asking as I wasn't involved in the early days of dual-connection growth so I'm trying to get my eyes around it.

Reply:
> It's not for now(all goods) but I wonder whether You considered such an approach for dual-connection lib?

No, but feel free to contribute.

> Asking as I wasn't involved in the early days of dual-connection growth

We are still in the early days. We didn't reach the 1.0.0 release yet. 😉